README.md

```diff
@@ -207,4 +207,23 @@ see the [Scaling Lightning](https://github.com/scaling-lightning/scaling-lightni
 project.
 
 ## Docker
-If you want to run the cli in a containerized environment, see the docker set up docs [here](./docker/README.md)
\ No newline at end of file
+If you want to run the cli in a containerized environment, see the docker set up docs [here](./docker/README.md)
+
+## Contributing
+SimLN is open and welcomes contribution from the community. When creating a pull request, ensure that you adhere to the following:
+### Passing tests
+All tests must pass. Do so by running the command
+```sh
+$ cargo test
+```
+### Check code
+Formatting and linting should adhere to the rules specified by default and/or specifically in `rustfmt.toml`. Adhere to the linter suggestions made by `clippy`.
+```sh
+$ make check-code
+```
+### Clean commit history
+- Have a clean commit history: 
+  - [Fixup and squash](https://andrewlock.net/smoother-rebases-with-auto-squashing-git-commits/) if you have to
```

Comment:
"If you have to" .. address review feedback?

I'd also make a note that it's preferable to push with fixups after the commit that they apply to and then squash once review is complete?